streams/runners/storm/runtime/src/test/java/com/hortonworks/streamline/streams/runtime/storm/bolt/rules/FunctionsTest.java

```diff
@@ -63,6 +65,22 @@ public void testFunctions() throws Exception {
     }
 
     @Test
+    public void testStringFunctions2() throws Exception {
+        doTest(readFile("/streamline-udf2.json"), getTuple());
+        new Verifications() {
+            {
+                String streamId;
+                Tuple anchor;
+                List<List<Object>> tuples = new ArrayList<>();
+                mockCollector.emit(streamId = withCapture(), anchor = withCapture(), withCapture(tuples));
+                Assert.assertEquals(1, tuples.size());
+                Assert.assertEquals("SUBSTRING ello SUBSTRING2 hell POSITION 4 POSITION2 0 TRIM space LTRIM space  RTRIM  space OVERLAY abba OVERLAY2 abbaa",
```

Comment:
The result of POSITION2 should be 4 after addressing bug I commented. If you would want to test about 'none matched', set parameter to 5.